### ISSUE TITLE
[refmt] [vscode] Saving a newly empty file repopulates it with the last save

### ISSUE THREAD

**Issue by ubsan** [facebook/reason#1680]
(I don't know if this is a vscode or refmt issue)

To reproduce:

Have a non-empty .re file. Delete all text in the file. Save.

**Comment by chenglou** [comment]
That might be a ocaml-language-server race condition? cc @freebroccolo @jchavarri

**Comment by jchavarri** [comment]
I can't repro. @ubsan i guess you're on v1.0.16 of the vscode Reason extension, right? Are you on Windows? WSL?

Could you check if the problem still appears if you run refmt manually? (by using alt-shift-F instead of saving). Also could you post what value do you have for `reason.diagnostics.tools` in vscode settings? (if any)

**Comment by jchavarri** [comment]
@chenglou I think @freebroccolo fixed the race condition in 1.0.16 by disabling merlin auto-restart temporarily (it'll be re-enabled in the next version).

**Comment by ubsan** [comment]
@jchavarri yeah, I'm on v1.0.16, and on Windows WSL, and it does still happen if I run refmt manually.

`reason.diagnostics.tools = []`

**Comment by freebroccolo** [comment]
I don't think this should be related to any of the merlin auto-restart functionality. I haven't encountered anything like this as far as I can recall but I'll investigate.

**Comment by ubsan** [comment]
Somebody else on the discord, brentbaum, confirmed the bug.

**Comment by freebroccolo** [comment]
Okay, I think I know what might be causing this. Should have a fix in the next release.

**Comment by ubsan** [comment]
cool :)

**Comment by thangngoc89** [comment]
I can confirm this bug. I’m on Ubuntu and something, the editor keep adding the deleted line. The only way I can fix it is to restart the editor

**Comment by FallenMax** [comment]
Any update? This also happens on my setup:

OS: Ubuntu 16.04
VSCode: 1.19.2
refmt: Reason 3.0.4 @ 731e605

**Comment by thangngoc89** [comment]
@FallenMax We should all tracking the issue here https://github.com/reasonml-editor/vscode-reasonml/issues/156 

This is a VS Code's plugin bug

**Comment by freebroccolo** [comment]
This should be fixed in the vscode extension version `1.0.23`.
